bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/caster/SagerWeatherCaster.java

```diff
@@ -319,27 +320,32 @@ private void updatePrediction() {
                 }
         }
         String forecast = forecaster.getProperty(d1 + sagerPressure + pressureEvolution + nubes);
-        prevision = Optional.ofNullable(forecast != null ? new SagerPrediction(forecast) : null);
+        prevision = forecast != null ? new SagerPrediction(forecast) : null;
     }
 
     public String getForecast() {
-        return prevision.map(p -> p.getForecast()).orElse(UNDEF);
+        SagerPrediction prevision = this.prevision;
+        return prevision != null ? prevision.getForecast() : UNDEF;
     }
 
     public String getWindVelocity() {
-        return prevision.map(p -> p.getWindVelocity()).orElse(UNDEF);
+        SagerPrediction prevision = this.prevision;
+        return prevision != null ? prevision.getWindVelocity() : UNDEF;
     }
 
     public String getWindDirection() {
-        return prevision.map(p -> p.getWindDirection()).orElse(UNDEF);
+        SagerPrediction prevision = this.prevision;
+        return prevision != null ? prevision.getWindDirection() : UNDEF;
     }
 
     public String getWindDirection2() {
-        return prevision.map(p -> p.getWindDirection2()).orElse(UNDEF);
+        SagerPrediction prevision = this.prevision;
+        return prevision != null ? prevision.getWindDirection2() : UNDEF;
     }
 
     public String getSagerCode() {
-        return prevision.map(p -> p.getSagerCode()).orElse(UNDEF);
+        SagerPrediction prevision = this.prevision;
+        return prevision != null ? prevision.getSagerCode() : UNDEF;
```

Comment:
You can probably reduce the lines of code using `Objects.requireNonNullElse` in this file too.

Reply:
Unfortunately not. Calling a method on a nullable is generating a compiler warning. Or do you mean something differnet then:
`return Objects.requireNonNullElse(this.prevision.getForecast(), UNDEF);`

Reply:
Yes you're right! Didn't look closely. :detective: